### PR TITLE
De underclock f3

### PR DESCRIPTION
**Pull request by Schalonsus**
Enabling underclock for STM32F10 targets, disabling some unused features on most F1 and enabling F1 for compiling.
Code compiled fine, but not tested yet

### CODE REVIEW

File: src/main/fc/cli.c [hunk 1]
```diff
@@ -988,6 +988,9 @@ static const clivalue_t valueTable[] = {
 #if defined(STM32F3)
     { "cpu_underclock",             VAR_INT8   | MASTER_VALUE | MODE_LOOKUP, .config.lookup = { TABLE_OFF_ON }, PG_SYSTEM_CONFIG, offsetof(systemConfig_t, cpuUnderclock) },
 #endif
+#if defined(STM32F1)
```
Comment (digitalentity) [inline src/main/fc/cli.c]: This can be merged to above: ` #if defined(STM32F1) || defined(STM32F3)`

File: src/main/fc/fc_init.c [hunk 2]
```diff
@@ -191,6 +191,10 @@ void init(void)
     systemClockSetup(systemConfig()->cpuUnderclock);
 #endif
 
+#ifdef STM32F1
```
Comment (digitalentity) [inline src/main/fc/fc_init.c]: This can be merged to above: `#if defined(STM32F1) || defined(STM32F3)`

**Review by digitalentity — CHANGES_REQUESTED** [review]
Please refactor according to requested changes and test. Overall - looks good :+1:

**Comment by Schalonsus** [comment]
Compiled, flashed and tested on the bench.
Will make a test flight the next days.

File: Makefile [hunk 3]
```diff
@@ -16,7 +16,7 @@
 #
 
 # The target to build, see VALID_TARGETS below
-TARGET    ?= SPRACINGF3
+TARGET    ?= NAZE
```
Comment (digitalentity) [inline Makefile]: Please don't change back to default NAZE

**Review by digitalentity — CHANGES_REQUESTED** [review]
Please don't enable NAZE target back. We'll merge this feature for custom firmware builders, but it's unlikely that NAZE will make it to next release. It might if we save some space, but that's not guaranteed.

**Comment by digitalentity** [comment]
Cool! Let me know about outcome of the test-flight - I'm curious if it made a difference.

**Comment by Schalonsus** [comment]
If i use this hex, receiver input is not working anymore

**Review by digitalentity — APPROVED** [review]

**Comment by digitalentity** [comment]
@Schalonsus on what board?

**Comment by Schalonsus** [comment]
Naze32, this even happen if i just apply the underclock code for F1/F3 to the latest master
I tested latest master 1.7.2 with underclock code and de_underclock branch 1.7.3

**Comment by digitalentity** [comment]
Likely clock setup code is missing something critical. What's the output of `status` command on Naze with and without underclock option enabled?

**Comment by Schalonsus** [comment]
with underclock:

CLI

# status
System Uptime: 14 seconds
Voltage: 0 * 0.1V (3S battery - OK)
CPU Clock=72MHz, GYRO=MPU6050, ACC=MPU6050.n, BARO=BMP085
Sensor status: GYRO=OK, ACC=OK, MAG=NONE, BARO=OK, RANGEFINDER=NONE, GPS=UNAVAILABLE
Stack size: 2048, Stack address: 0x20005000
I2C Errors: 0, config size: 1410, max available config: 2048
ADC channel usage:
   BATTERY : configured = ADC 2, used = none
      RSSI : configured = ADC 3, used = none
   CURRENT : configured = ADC 1, used = none
  AIRSPEED : configured = none, used = none
System load: 5, cycle time: 2005, PID rate: 498, RX rate: 49, System rate: 9

# 


without underclock:

CLI

# status
System Uptime: 9 seconds
Voltage: 0 * 0.1V (3S battery - OK)
CPU Clock=48MHz, GYRO=MPU6050, ACC=MPU6050.n, BARO=BMP085
Sensor status: GYRO=OK, ACC=OK, MAG=NONE, BARO=OK, RANGEFINDER=NONE, GPS=NONE
Stack size: 2048, Stack address: 0x20005000
I2C Errors: 0, config size: 1410, max available config: 2048
ADC channel usage:
   BATTERY : configured = ADC 2, used = none
      RSSI : configured = ADC 3, used = none
   CURRENT : configured = ADC 1, used = none
  AIRSPEED : configured = none, used = none
System load: 322, cycle time: 2070, PID rate: 483, RX rate: 30, System rate: 9

#

**Comment by digitalentity** [comment]
I've merged this - will fix in that branch

**Pull request merged**